minetest/minetest
# Allow toggling touchscreen mode at runtime

#### What

Allow runtime switching between desktop and touchscreen mode.

#### How

This PR replaces hardcoded `#ifdef`s with setting (`enable_touch`).

#### Resolves:
 - Fixes (but just partly) https://github.com/minetest/minetest/issues/10062
 - Fixes https://github.com/minetest/minetest/issues/12785

#### Problem solved by this:

Ability to distribute one binary for touchscreen enabled and classic (desktops and laptops without touchscreens).

https://floss.social/@okias/111417351641760633

People love to play Minetest on Linux phones. Some of these phones are convergent, so they work as desktop when connected to USB-C.

Also we have Linux tablets and convertible PCs.


## How to test

In the settings appear new checkbox for the touchscreen. You can check it or uncheck it and see whats going to happen.

#### TODO
 - At some point would be lovely to have touchscreen getting auto-detected.

#### Depends
 - on 5.9 (currently `master` branch), for code applicable on 5.8, use https://github.com/okias/minetest/tree/linux-touch-5.8
 - https://github.com/minetest/minetest/pull/14117
 - https://github.com/minetest/minetest/pull/14146
 - https://github.com/minetest/minetest/pull/13282
